react-app/src/components/SaveOutputForm.js

```diff
@@ -0,0 +1,213 @@
+        />
+      );
+    });
+    return list;
+  }
+
+  function toggleSelectAll() {
+    // modify which files will be saved first, so that there's no issues with state update delays
+    if (!selectAll) {
+      props.extractedData.forEach((bundle, i) => {
+        const label = getLabel(bundle, i);
+        defaultWhichFiles = {
+          ...defaultWhichFiles,
+          [label]: true,
+        };
+      });
+      setWhichFiles({ ...defaultWhichFiles });
+    }
+    // set select all
+    setSelectAll(!selectAll);
+  }
+
+  function onSave() {
+    const outputBundles = [];
+    props.extractedData.forEach((bundle, i) => {
+      if (whichFiles[getLabel(bundle, i)] === true) {
+        outputBundles.push({ bundle, index: i });
+      }
+    });
+    if (outputBundles.length > 0) {
+      window.api
+        .saveOutput(outputPath, outputBundles, props.loggedMessages, saveLogs)
+        .then((result) => {
+          if (result === true) {
+            // if saveOutput() returns true, then the save process succeeded
+            setShowSavedAlert(true);
+          }
+        })
+        .catch((error) => {
+          setShowErrorAlert(true);
+          setErrorMessage(error.message);
+          setShowSavedAlert(false);
+          setShowNoFilesAlert(false);
+        });
+    } else {
+      setShowNoFilesAlert(true);
+    }
+  }
+  function onCancel() {
+    props.setShowSaveForm(false);
+  }
+  return (
+    <div className="page-container">
+      <h3 className="page-subtitle">Save Output</h3>
+      <div className="page-container">
+        <Form className="form-container">
+          <Row>
+            <Col>
+              <Form.Group controlId="formConfigPath" className="mb-3">
+                <Form.Label className="form-label">Select Output Folder</Form.Label>
+                <div className="file-picker-box">
+                  <div className="file-button-container">
+                    <Button className="generic-button narrow-button" variant="outline-info" onClick={onSetOutputPath}>
+                      Select Folder
+                    </Button>
+                    <Button className="generic-button narrow-button" variant="outline-info" onClick={clearOutputPath}>
+                      Clear
+                    </Button>
+                  </div>
+                  <Form.Label className="form-label file-name">{outputPath}</Form.Label>
+                </div>
+              </Form.Group>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              <Form.Group controlId="formConfigPath" className="mb-3">
+                <Form.Check type="checkbox" label="Save logger messages" checked={saveLogs} onChange={toggleSaveLogs} />
+              </Form.Group>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              <Form.Group controlId="formConfigPath" className="mb-3">
+                <Form.Label className="form-label">Select Files to Save</Form.Label>
```

Comment:
@Dtphelan1 noticed this bug, so I'm giving him credit for it. If you click on the text "Select All", not the checkbox itself, it actually selects the "Save logger messages" box above. This should work the same as clicking on the patient names does and check the "Select All" box.

Reply:
I have fixed this bug.